samila/genimage.py

```diff
@@ -32,32 +32,40 @@ def __init__(self, function1=None, function2=None, data=None):
         :type function2: python or lambda function
         :param data: prior generated data
         :type data: (io.IOBase & file)
+        :param config: generative image configs
+        :type config: JSON
         """
+        _GI_initializer(self)
         self.function1 = function1
-        self.function2 = function2
         self.function1_str = None
+        self.function2 = function2
         self.function2_str = None
         self.fig = None
-        if function1 is None or function2 is None:
-            if data is None:
-                self.function1_str = random_equation_gen()
-                self.function2_str = random_equation_gen()
-                self.function1 = eval("lambda x,y:" + self.function1_str)
-                self.function2 = eval("lambda x,y:" + self.function2_str)
-            else:
-                warn(JUST_DATA_WARNING, RuntimeWarning)
+        if config is not None:
+            matplotlib_version = load_config(self, config)
+            if matplotlib_version != matplotlib.__version__:
```

Comment:
I think here we have duplication, I suggest adding `matplot_version` as an attribute to the `GenerativeImage` object (default value: `matplotlib.__version__`) and checking it once.